label_studio/core/views.py

```diff
@@ -169,6 +170,21 @@ def samples_paragraphs(request):
     return HttpResponse(json.dumps(result), content_type='application/json')
 
 
+def heidi_tips(request):
+    """Fetch live tips from github raw liveContent.json to avoid caching and client side CORS issues"""
+    url = 'https://raw.githubusercontent.com/HumanSignal/label-studio/refs/heads/develop/web/apps/labelstudio/src/components/HeidiTips/liveContent.json'
+    response = requests.get(
+        url,
+        headers={'Cache-Control': 'no-cache', 'Content-Type': 'application/json', 'Accept': 'application/json'},
+        timeout=5,
+    )
+
+    # Raise an exception for bad status codes to avoid caching
+    response.raise_for_status()
```

Comment:
In that case it just uses whatever loaded most recent in localStorage, and if never loaded from the endpoint it goes to the default which is what we were already using.

Reply:
We won't block for errors, or even timeouts. I put a strict timeout on the call as I didn't want this loading slow or showing nothing ever. I wanted it to just get and update the data as naturally as possible where the user doesn't notice it's happening at all. As well, for airgapped environments we cannot make that call anyways so it will fail fast and the defaults are all that will be provided. Now that I explain this all, I just had a thought to make this even better 👍.